Chase

ChaseC1998 opened the pull request:
I updated my branch with the master content so all of the stuff just before this is still there. I just have @Override before the attack method in dungeoncharacter class. Don't think the interface is correct but im confused how else to do it with the inheritance hierarchy we have. Here is just what i have now and everything still functions as before.

ChaseC1998 committed:
Attempting proper attack interface

I dont know if this is right at all. The interface is implemented into dungeon character and the attack method extends the interface but I dont know if that actually makes any difference. The code still runs the same and doesnt have any problems but I don't think the interface is doing anything.

ChaseC1998 committed:
Merge branch 'master' into Chase

AlexD4612 reviewed (changes requested):
What you could also do is combine all 3 of the heroes special abilities into 1 method in the interface, maybe call it "special ability" and instead of having sorceress call increaseHitPoints, she will call the special ability, and same for the thief's surprise attack and warrior's crushing blow, they would of course each be defined inside the subclasses as opposed to the DungeonCharacter class because they are different for each hero

ChaseC1998 committed:
Properly implement the attack interface

It's actually in how I think it's supposed to be. The hero subclasses had their special ability methods removed and replaced with an override of the interface one.
